src/Microsoft.Azure.SignalR.Protocols/ServiceProtocol.cs

```diff
@@ -661,7 +661,16 @@ private static void WriteAckMessage(ref MessagePackWriter writer, AckMessage mes
         writer.Write(ServiceProtocolConstants.AckMessageType);
         writer.Write(message.AckId);
         writer.Write(message.Status);
-        writer.Write(message.Message);
+        if (message.Payload.HasValue)
```

Comment:
change to WriteArrayHeader(6) for this added property message.Payload instead?

Reply:
We cannot extend the array length, as in old protocol, we expect that the extension members start from the fifth...